feature/quicksettings/src/main/java/com/google/jetpackcamera/feature/quicksettings/QuickSettingsScreen.kt

```diff
@@ -0,0 +1,178 @@
+    onLensFaceClick: (lensFace: Boolean) -> Unit,
+    onFlashModeClick: (flashMode: FlashModeStatus) -> Unit,
+    //onAspectRatioClick: (aspectRatio: CameraAspectRatio) -> Unit,
+    //onTimerClick: (timer: CameraTimer) -> Unit
+) {
+    var shouldShowQuickSetting by remember {
+        mutableStateOf(IsExpandedQuickSetting.NONE)
+    }
+
+    val backgroundColor = animateColorAsState(
+        targetValue = Color.Black.copy(alpha = if (isOpen) 0.7f else 0f),
+        label = "backgroundColorAnimation"
+    )
+
+    val contentAlpha = animateFloatAsState(
+        targetValue = if (isOpen) 1f else 0f, label = "contentAlphaAnimation",
+        animationSpec = tween()
+    )
+
+    Column(
+        modifier = modifier
+            .fillMaxSize()
+            .background(color = backgroundColor.value),
+        verticalArrangement = Arrangement.Top,
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+        DropDownIcon(toggleIsOpen, isOpen = isOpen)
+
+        if (isOpen) {
+            Column(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .alpha(alpha = contentAlpha.value)
+                    .clickable {
+                        // if a setting is expanded, close it. if no other settings are expanded, then close out of the popup
+                        if (shouldShowQuickSetting == IsExpandedQuickSetting.NONE) toggleIsOpen() else shouldShowQuickSetting =
+                            IsExpandedQuickSetting.NONE
+                    },
+                verticalArrangement = Arrangement.Center,
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+                ExpandedQuickSettingsUi(
+                    currentCameraSettings = currentCameraSettings,
+                    shouldShowQuickSetting = shouldShowQuickSetting,
+                    setVisibleQuickSetting = { enum: IsExpandedQuickSetting ->
+                        shouldShowQuickSetting = enum
+                    },
+                    onLensFaceClick = onLensFaceClick,
+                    onFlashModeClick = onFlashModeClick,
+                    //onAspectRatioClick = onAspectRatioClick,
+                    //onTimerClick = onTimerClick,
+                )
+            }
+        }
+    }
+}
+
+// enum representing which individual quick setting is currently expanded
+private enum class IsExpandedQuickSetting {
+    NONE,
+    ASPECT_RATIO,
+}
+
+/**
+ * The UI component for quick settings when it is expanded.
+ */
+@Composable
+private fun ExpandedQuickSettingsUi(
+    currentCameraSettings: CameraAppSettings,
+    onLensFaceClick: (lensFacingFront: Boolean) -> Unit,
+    onFlashModeClick: (flashMode: FlashModeStatus) -> Unit,
+    shouldShowQuickSetting: IsExpandedQuickSetting,
+    setVisibleQuickSetting: (IsExpandedQuickSetting) -> Unit
+    // onAspectRatioClick: (aspectRatio: CameraAspectRatio) -> Unit,
+    // onTimerClick: (timer: CameraTimer) -> Unit,
+) {
+
+    Column(
```

Comment:
There used to be a width calculation algorithm here to make sure we fit the appropriate number of components in each row according to the screen width. I think we should use that instead of hardcoding the item to 3. This would be especially relevant in landscape mode if we ever support that

Reply:
I moved it into the `QuickSettingsGrid` component in `QuickSettingsComponents.kt`